jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildTarTask.java

```diff
@@ -0,0 +1,182 @@
+    // Asserts required @Input parameters are not null.
+    Preconditions.checkNotNull(jibExtension);
+    GradleBuildLogger gradleBuildLogger = new GradleBuildLogger(getLogger());
+    jibExtension.handleDeprecatedParameters(gradleBuildLogger);
+
+    RegistryCredentials knownBaseRegistryCredentials = null;
+    Authorization fromAuthorization = jibExtension.getFrom().getImageAuthorization();
+    if (fromAuthorization != null) {
+      knownBaseRegistryCredentials = new RegistryCredentials("jib.from.auth", fromAuthorization);
+    }
+
+    GradleProjectProperties gradleProjectProperties =
+        GradleProjectProperties.getForProject(getProject(), gradleBuildLogger);
+    String mainClass = gradleProjectProperties.getMainClass(jibExtension);
+    ImageReference targetImage =
+        gradleProjectProperties.getDockerTag(jibExtension, gradleBuildLogger);
+
+    // Builds the BuildConfiguration.
+    // TODO: Consolidate with BuildImageTask/BuildDockerTask.
+    BuildConfiguration.Builder buildConfigurationBuilder =
+        BuildConfiguration.builder(gradleBuildLogger)
+            .setBaseImage(ImageReference.parse(jibExtension.getBaseImage()))
+            .setTargetImage(targetImage)
+            .setBaseImageCredentialHelperName(jibExtension.getFrom().getCredHelper())
+            .setKnownBaseRegistryCredentials(knownBaseRegistryCredentials)
+            .setMainClass(mainClass)
+            .setJavaArguments(jibExtension.getArgs())
+            .setJvmFlags(jibExtension.getJvmFlags())
+            .setExposedPorts(ExposedPortsParser.parse(jibExtension.getExposedPorts()))
+            .setAllowHttp(jibExtension.getAllowInsecureRegistries());
+    if (Files.exists(jibExtension.getExtraDirectory().toPath())) {
+      try (Stream<Path> extraFilesLayerDirectoryFiles =
+          Files.list(jibExtension.getExtraDirectory().toPath())) {
+        buildConfigurationBuilder.setExtraFilesLayerConfiguration(
+            LayerConfiguration.builder()
+                .addEntry(extraFilesLayerDirectoryFiles.collect(Collectors.toList()), "/")
+                .build());
+      }
+    }
+    CacheConfiguration applicationLayersCacheConfiguration =
+        CacheConfiguration.forPath(gradleProjectProperties.getCacheDirectory());
+    buildConfigurationBuilder.setApplicationLayersCacheConfiguration(
+        applicationLayersCacheConfiguration);
+    if (jibExtension.getUseOnlyProjectCache()) {
+      buildConfigurationBuilder.setBaseImageLayersCacheConfiguration(
+          applicationLayersCacheConfiguration);
+    }
+    BuildConfiguration buildConfiguration = buildConfigurationBuilder.build();
+
+    // TODO: Instead of disabling logging, have authentication credentials be provided
+    GradleBuildLogger.disableHttpLogging();
+
+    RegistryClient.setUserAgentSuffix(USER_AGENT_SUFFIX);
+
+    // Uses a directory in the Gradle build cache as the Jib cache.
+    try {
+      BuildStepsRunner.forBuildTar(
```

Comment:
Question: does Gradle not track artifacts resulting from a task? If so, would it be worth adding the tar as a build artifact?

Reply:
https://docs.gradle.org/current/userguide/artifact_management.html, so there is something about artifact management here, but I'm not sure exactly how it translate to what we're building - give our tars are container-images not java-artifacts. Still worth exploring though.